x-pack/solutions/security/plugins/security_solution/public/common/components/charts/draggable_legend_item.tsx

```diff
@@ -90,6 +107,24 @@ const DraggableLegendItemComponent: React.FC<{
             )}
           </EuiFlexGroup>
         </EuiFlexItem>
+
+        {isInlineActions && (
+          <EuiFlexItem grow={false} data-test-subj="legendItemInlineActions">
+            <SecurityCellActions
+              mode={CellActionsMode.INLINE}
+              visibleCellActions={0}
+              triggerId={SecurityCellActionsTrigger.DEFAULT}
+              data={{ field, value }}
+              sourcererScopeId={sourcererScopeId}
+              metadata={{
+                scopeId,
+                extraActionsIconType: 'boxesVertical',
+                extraActionsColor: 'text',
+              }}
+              disabledActionTypes={[SecurityCellActionType.SHOW_TOP_N]}
+            />
```

Comment:
I am not sure these new `extraActionsIconType` and `extraActionsColor` should be passed within the _metadata_ object. They are component properties that customize the presentation, rather than a kind of "data."

Have considered adding them as regular (optional) props?

Reply:
Thanks @semd! I started with them as props but noticed quite a bit of prop drilling to get to the actual extra action components. Hence taking advantage of the metadata and context to make it look cleaner. I thought the metadata was meant to be a 'catch all' usage, but good to know!

Anyways, I've updated them as props in [this commit](78a47c0bd158113a93dde32a5b1aa0b6c9d421f5), let me know what you think!

Reply:
Yes, _metadata_ is used to pass extra information about the cell, the metadata is accessible from the actions execution, so it's an easy way to change the behaviour of the actions. 
This property allows the cell actions to be more flexible and simplifies the implementation for corner cases. But it also has its downsides, it's not properly typed, and the nested values are not memoized, so we should not overuse it. 
For properties that are not "extra data" used inside action execution, like presentational properties like these, it's better to create separate props.

Thanks for applying the change 💯